pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -197,6 +199,21 @@ func (w *worker) deploy(ctx context.Context, operation string) (extensionsv1alph
 			workerPoolKubernetesVersion = *workerPool.Kubernetes.Version
 		}
 
+		// initializing nodeTemplate by fetching details from cloudprofile
+		var nodeTemplate extensionsv1alpha1.NodeTemplate
+		machineDetails := gardencorev1beta1helper.FindMachineTypeByName(w.values.CloudProfileMachineTypes, workerPool.Machine.Type)
+		if machineDetails == nil {
+			return nil, fmt.Errorf("unable to initialize nodeTemplate for workerPool %q, machine type %q not found in cloudprofile", workerPool.Name, workerPool.Machine.Type)
+		} else {
```

Comment:
Remove `else` block to reduce nesting (not needed since you `return` anyways).

Reply:
pls refer 2ea874669d2ae21a721ca289671785cfcfbe9948